### PR TITLE
Move clear link outside of Faceted Search navigation button

### PR DESCRIPTION
**Pull request by mattcoy-arcticleaf**
#### What?

The current HTML markup for the link to clear faceted search element's items is invalid as it is located inside a button. This update moves the clear link outside the navigation button, so it no longer violates W3C markup standards (see below for a screenshot of the error message). This update also adds padding so the title of the filter does not overlap with the clear button.

Additionally, this update componentizes the faceted search navigation. Previously, the faceted search navigation code was repeated in the exact same code across 3 different templates.

Lastly, this update cleans up some extraneous whitespace (for consistency with the rest of the codebase) and otherwise questionable code.

The only visible change to the theme is, as stated above, the padding adjustment to prevent the overlap of the faceted search filter title and the clear button.

#### Tickets / Documentation

N/A

#### Screenshots (if appropriate)

W3C Validator error message

### CODE REVIEW

**Comment by junedkazi** [comment]
@bigcommerce/artemis-dt

File: templates/components/faceted-search/facets/hierarchy.html [hunk 1]
```diff
@@ -46,7 +27,7 @@ <h5 class="accordion-title">
         </ul>
 
         {{#if show_more_toggle}}
-            <a href="#facetedSearch-navList--{{#replace '&' (hyphenate facet) }}{{else}}{{hyphenate facet}}{{/replace}}" role="button" class="toggleLink">
+            <a href="#facetedSearch-navList--{{dashcase facet}}" role="button" class="toggleLink">
```
Comment (junedkazi) [inline templates/components/faceted-search/facets/hierarchy.html:30]: @lord2800 will this break things for facets which have `&` ?

Comment (mattcoy-arcticleaf) [reply templates/components/faceted-search/facets/hierarchy.html:30]: @junedkazi , Nope! IMO, dashcase is the correct helper to use here as it strips special characters and converts everything into a nice, usable form.

**Comment by junedkazi** [comment]
@mattcoy-arcticleaf this is good to go. We need to hold off on merge because we are in the process of doing a release. As soon as that is rolled out we will merge this one.

**Comment by mattcoy-arcticleaf** [comment]
@junedkazi no worries. Do you want me to make a changelog entry?

File: templates/components/faceted-search/facets/hierarchy.html (hunk 1 above)
Comment (lord2800) [reply templates/components/faceted-search/facets/hierarchy.html:30]: Yeah, I wish I'd found that when we were initially looking into that problem. 😢

**Pull request merged**